### PR TITLE
[unit-tests-only] add unit tests for new collaborator component

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add unit test for new collaborator component

## Related Issue

- Related https://github.com/owncloud/web/issues/5234

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/18418/12/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/NewCollaborator.spec.js [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+
+    it('should render the share button', () => {
+      const wrapper = getShallowMountedWrapper({ user: 'user0' })
+      const shareButton = wrapper.find(selectors.shareButton)
+      expect(shareButton.exists()).toBe(true)
+      expect(shareButton.attributes().disabled).toBe('true')
+      expect(shareButton.text()).toBe('Share')
+    })
+
+    it('cancel button should work', async () => {
```
Comment (saw-jan) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/NewCollaborator.spec.js:313]:
```suggestion
    it('should call "$_ocCollaborators_newCollaboratorsCancel" method if cancel button is clicked', async () => {
```
or something better

File: packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/NewCollaborator.spec.js [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+
+      const wrapper = getMountedWrapper({ user: 'user0' })
+      const cancelButton = wrapper.find(selectors.cancelButton)
+
+      expect(cancelTrigger).not.toHaveBeenCalled()
+      await cancelButton.trigger('click')
+      expect(cancelTrigger).toHaveBeenCalledTimes(1)
+    })
+
+    it('save button should work', async () => {
```
Comment (saw-jan) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/NewCollaborator.spec.js:327]: same as above

File: packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/NewCollaborator.spec.js [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+      await new Promise(resolve => {
+        setTimeout(() => {
+          const select = wrapper.findComponent(VueSelect)
+          expect(select.props('options')).toMatchObject([])
+          resolve()
+        }, 3000)
+      })
+    })
+
+    it('Users should be listed in the autocomplete dropdown', async () => {
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/NewCollaborator.spec.js:125]: starting scenario title with `should` would be nicer and uniform 
```suggestion
    it('should display users list in the autocomplete dropdown', async () => {
```

**Comment by ownclouders** [comment]
Results for <strong>oCISResharing1</strong> https://drone.owncloud.com/owncloud/web/19129/60/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/19129/52/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingBasic</strong> https://drone.owncloud.com/owncloud/web/19154/50/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/19154/47/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/19154/52/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/19154/53/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISRename</strong> https://drone.owncloud.com/owncloud/web/19154/49/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPermissions1</strong> https://drone.owncloud.com/owncloud/web/19156/57/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/19156/53/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/19156/47/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalGroups</strong> https://drone.owncloud.com/owncloud/web/19156/54/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by individual-it** [comment]
blocked waiting for review @kulmann @pascalwengerter

**Comment by pascalwengerter** [comment]
Closed in favor of #6076, sorry for being unresponsive so long. The sidebar sharing components are receiving a major overhaul in #6039, I'll see what I can re-use from your work here - thanks!

**Pull request closed without merging**